packages/flutter/lib/src/widgets/sensitive_content.dart

```diff
@@ -0,0 +1,276 @@
+    }
+    if (_autoSensitiveWidgetCount > 0) {
+      return ContentSensitivity.autoSensitive;
+    }
+    if (_notSensitiveWigetCount > 0) {
+      return ContentSensitivity.notSensitive;
+    }
+    return null;
+  }
+}
+
+/// Host of the current content sensitivity level for the widget tree that contains
+/// some number [SensitiveContent] widgets.
+class SensitiveContentHost {
+  SensitiveContentHost._();
+
+  bool? _contentSenstivityIsSupported;
+  ContentSensitivitySetting? _contentSensitivitySetting;
+  ContentSensitivity? _defaultContentSensitivitySetting;
+
+  final SensitiveContentService _sensitiveContentService = SensitiveContentService();
+
+  /// The current [ContentSensitivity] level set for the entire widget tree.
+  @visibleForTesting
+  ContentSensitivity? currentContentSensitivityLevel;
+
+  /// [SensitiveContentHost] instance for the widget tree.
+  @visibleForTesting
+  static final SensitiveContentHost instance = SensitiveContentHost._();
+
+  /// The state of content sensitivity in the widget tree.
+  ///
+  /// Contains the number of widgets with each [ContentSensitivity] level and
+  /// the current [ContentSensitivity] setting.
+  @visibleForTesting
+  ContentSensitivitySetting? getContentSenstivityState() {
+    return _contentSensitivitySetting;
+  }
+
+  /// Registers a [SensitiveContent] widget that will help determine the
+  /// [ContentSensitivity] level for the widget tree.
+  static Future<void> register(ContentSensitivity desiredSensitivityLevel) {
+    return instance._register(desiredSensitivityLevel);
+  }
+
+  Future<void> _register(ContentSensitivity desiredSensitivityLevel) async {
+    _contentSenstivityIsSupported ??= await _sensitiveContentService.isSupported();
+    if (!_contentSenstivityIsSupported!) {
+      // Setting content sensitivity is not supported on this device.
+      return;
+    }
+
+    // If needed, set default content sensitivity level as set in native Android. This will be
+    // auto sensitive if it is otherwise unset by the developer. Also, initialize the current
+    // content sensitivity level if needed.
+    _defaultContentSensitivitySetting ??= await _sensitiveContentService.getContentSensitivity();
+    currentContentSensitivityLevel ??= _defaultContentSensitivitySetting;
+
+    // If needed, then set the initial content sensitivity state.
+    _contentSensitivitySetting ??= ContentSensitivitySetting();
+
+    // Update SensitiveContent widget count for those with desiredSensitivityLevel.
+    _contentSensitivitySetting!.addWidgetWithContentSensitivity(desiredSensitivityLevel);
+
+    // Verify that desiredSensitivityLevel should be set in order for sensitive
+    // content to remain obscured.
+    if (!shouldSetContentSensitivity(desiredSensitivityLevel)) {
+      return;
+    }
+
+    // Set content sensitivity level as desiredSensitivityLevel.
+    try {
+      _sensitiveContentService.setContentSensitivity(desiredSensitivityLevel);
```

Comment:
@justinmc I added try/catch blocks to the `setContentSensitivity` calls to make it clear that if this fails for some reason, we won't update the current setting. They should just catch the `FlutterError`s that I throw upon an error coming over the method channel. PTAL.

Reply:
I think you can just use a comment to explain that.